docs/USAGE-GUIDE.md

```diff
@@ -603,6 +603,56 @@ a pointer to abitrary data for use within the callback, a pointer to a key
 which can be used to delete the cached entry, and a 64 bit unsigned integer
 specifying the size of this key.
 
+
+## Record sizes
+
+### Throughput vs Latency
+
+When sending data, s2n-tls uses a default maximum record size which experimentation
+has suggested provides a reasonable balance of performance and throughput.
+
+**s2n_connection_prefer_throughput** can be called to increase the record size, which
+minimizes overhead. It also increases s2n-tls's memory usage.
+
+**s2n_connection_prefer_low_latency** can be called to decrease the record size, which
+allows the receiver to decrypt the data faster. It also decreases s2n-tls's memory usage.
+
+These options only affect the size of the records that s2n-tls sends, not the behavior
+of the peer.
+
+### Maximum Fragment Length
+
+The maximum number of bytes that can be sent in a TLS record is called the "maximum fragment length",
+and is set to 2^14 bytes by default. Regardless of the maximum record size that s2n-tls
+uses when sending, it may receive records containing up to 2^14 bytes of plaintext.
+
+A client can request a lower maximum fragment length by calling **s2n_config_send_max_fragment_length**,
+reducing the size of TLS records sent and providing benefits similar to **s2n_connection_prefer_low_latency**.
+However, many TLS servers either ignore these requests or handle them incorrectly, so a client should
+never assume that a lower maximum fragment length will be honored. If a server accepts the requested
+maximum fragment length, the client will respect that maximum when sending.
+
+By default, an s2n-tls server will ignore a client's requested maximum fragment length.
+If **s2n_config_accept_max_fragment_length** is called, the server will respect the client's requested
+maximum fragment length when sending, but will not reject client records with a larger fragment size.
+
+If a maximum fragment length is negotiated during the connection, it will override the behavior
+configured by **s2n_connection_prefer_throughput** and **s2n_connection_prefer_low_latency**.
+
+### Dynamic Record Sizing
+
+**s2n_connection_set_dynamic_record_threshold** can be called to change the record size dynamically.
```

Comment:
After reading this and the s2n.h comments, I still feel like I don't know why you would want to use this API. Can you elaborate on a use-case for this API?